### ISSUE TITLE
two isolate send message by the port will drop some messges of flutter version 1.9

### ISSUE THREAD

**Issue by ping996** [flutter/flutter#47208]
This is an issue of the dart isolate, I created a new isolate to send an int number to the main isolate,
the main isolate can't receive all the message send.

Dart core libraries ("dart:isolate".)
<details>
  <summary>code sample</summary>

```dart 
`import 'package:flutter/material.dart';
import 'dart:isolate';
import 'dart:async';
void main() {
init();
runApp(MyApp());
}

void test(SendPort port) {
var cnt = 1;
Timer.periodic(new Duration(microseconds: 1), (Timer timer){
port.send(cnt++);
});
}

void init() {
ReceivePort port = new ReceivePort();
print("create an isoalte...");
Isolate.spawn(test, port.sendPort);
port.listen((msg){
print("the return is: $msg");
});
}

class MyApp extends StatelessWidget {
// This widget is the root of your application.
@override
Widget build(BuildContext context) {
return MaterialApp(
title: 'Flutter Demo',
theme: ThemeData(
// This is the theme of your application.
//
// Try running your application with "flutter run". You'll see the
// application has a blue toolbar. Then, without quitting the app, try
// changing the primarySwatch below to Colors.green and then invoke
// "hot reload" (press "r" in the console where you ran "flutter run",
// or simply save your changes to "hot reload" in a Flutter IDE).
// Notice that the counter didn't reset back to zero; the application
// is not restarted.
primarySwatch: Colors.blue,
),
home: MyHomePage(title: 'Flutter Demo Home Page'),
);
}
}

class MyHomePage extends StatefulWidget {
MyHomePage({Key key, this.title}) : super(key: key);

// This widget is the home page of your application. It is stateful, meaning
// that it has a State object (defined below) that contains fields that affect
// how it looks.

// This class is the configuration for the state. It holds the values (in this
// case the title) provided by the parent (in this case the App widget) and
// used by the build method of the State. Fields in a Widget subclass are
// always marked "final".

final String title;

@override
_MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State {
int _counter = 0;
void _incrementCounter() {
setState(() {
// This call to setState tells the Flutter framework that something has
// changed in this State, which causes it to rerun the build method below
// so that the display can reflect the updated values. If we changed
// _counter without calling setState(), then the build method would not be
// called again, and so nothing would appear to happen.
_counter++;
});
}

@override
Widget build(BuildContext context) {
// This method is rerun every time setState is called, for instance as done
// by the _incrementCounter method above.
//
// The Flutter framework has been optimized to make rerunning build methods
// fast, so that you can just rebuild anything that needs updating rather
// than having to individually change instances of widgets.
return Scaffold(
appBar: AppBar(
// Here we take the value from the MyHomePage object that was created by
// the App.build method, and use it to set our appbar title.
title: Text(widget.title),
),
body: Center(
// Center is a layout widget. It takes a single child and positions it
// in the middle of the parent.
child: Column(
// Column is also a layout widget. It takes a list of children and
// arranges them vertically. By default, it sizes itself to fit its
// children horizontally, and tries to be as tall as its parent.
//
// Invoke "debug painting" (press "p" in the console, choose the
// "Toggle Debug Paint" action from the Flutter Inspector in Android
// Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
// to see the wireframe for each widget.
//
// Column has various properties to control how it sizes itself and
// how it positions its children. Here we use mainAxisAlignment to
// center the children vertically; the main axis here is the vertical
// axis because Columns are vertical (the cross axis would be
// horizontal).
mainAxisAlignment: MainAxisAlignment.center,
children: [
Text(
'You have pushed the button this many times:',
),
Text(
'$_counter',
style: Theme.of(context).textTheme.display1,
),
],
),
),
floatingActionButton: FloatingActionButton(
onPressed: _incrementCounter,
tooltip: 'Increment',
child: Icon(Icons.add),
), // This trailing comma makes auto-formatting nicer for build methods.
);
}
}
```

</details>

the log:
```console
/flutter (10170): the return is: 21267
I/flutter (10170): the return is: 21283
I/flutter (10170): the return is: 21291
I/flutter (10170): the return is: 21305
I/flutter (10170): the return is: 21306
I/flutter (10170): the return is: 21309
I/flutter (10170): the return is: 21328
I/flutter (10170): the return is: 21345
```

does the isolate port will miss message if the message send quickly.

I also test this only in dart with the blew code:

```dart
import 'dart:async';
import 'dart:isolate';

void main() async {
  final port = ReceivePort();
  port.listen(print);
  Isolate.spawn(sendNumbers, port.sendPort);
}

void sendNumbers(SendPort sendPort) {
  var cnt = 0;
  Timer.periodic(const Duration(microseconds: 10), (_) {
    sendPort.send(cnt++);
  });
}
```

The main isolate can receive all the messages sent from the another one.

**Comment by iapicca** [comment]
Hi @ping996 
you mention 2 isolates, but I see only one in your example,
I quickly made a sample code to test if the data produced by 2 isolate

<details>
  <summary>sample</summary>

```dart
import 'dart:async';
import 'dart:isolate';
import 'package:flutter/material.dart';

void _runTimer(SendPort sendPort) {
  int _counter = 0;
  Timer.periodic(Duration(microseconds: 1), (Timer t) => sendPort.send(_counter +=1));
}

Future<List<int>> _listFromIsolate() async {
  final List<int> _list = [];
  final ReceivePort _receivePort = ReceivePort();
  final Isolate _isolate = await Isolate.spawn(_runTimer, _receivePort.sendPort);
  _receivePort.listen((data)=> _list.add(data as int));
  await Future.delayed(const Duration(seconds: 3));
  _isolate.kill();
  return _list;
}

bool _areListsIdentical(List<int> _list1, List<int> _list2) {
  if (_list1.length != _list2.length) return false;
  for (int i = 0; i < _list1.length; ++i) {
    if (_list1[i] != _list2[i]) return false;
  }
  return true;
}

bool _areIntProgressive(List<int> _list) {
  for (int i = 0; i <_list.length; ++i) {
    if (i<_list.length-1) {
      if(_list[i] != (_list[i+1]-1)) return false;
    }
  }
  return true;
}

Future<bool> get _test async {
  final List<int> _list1 = await _listFromIsolate();
  final List<int> _list2 = await _listFromIsolate();
  if (!_areListsIdentical(_list1,_list2)) return false;
  if (!_areIntProgressive(_list1)) return false;
  if (!_areIntProgressive(_list2)) return false;
  return true;
}


void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) =>  MaterialApp(
    debugShowCheckedModeBanner: false,
    home: MyHomePage(),
  );
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) => Scaffold(
    body: Center(
      child: FutureBuilder<bool>(
        initialData: null,
        future: _test,
        builder: (BuildContext context, AsyncSnapshot snapshot) => snapshot.hasData
          ? Text(snapshot.data?'failed':'succeded')
          : CircularProgressIndicator(),
  ),),);
}
```

</details>

I'm not able to reproduce the issue you are describing.
Are you experiencing this issue with the latest stable version?
thank you

**Comment by jason-simmons** [comment]
I tried reproducing this with the current Flutter master branch on a few Android devices, but I did not notice any lost messages.

What device and which version of Flutter are you using when you see this happen?

**Comment by ping996** [comment]
I use the stable of flutter 1.9 version and dart of 2.5, my logic are:
1, create the second isolate when main starts.
2, the second will sends two messages though the port
3, when the first message comes in the main isoalte, main isolate will do some logic of rendering
4, the second messges has droped of the main. from the dart code, the second isolate had send, but the main didn't received it, 

My test platform is the android 10, I havn't confirm whether the second isolate doesn't put the message successfully or the main isolate doesn't reiceve correctly?
It only occurs some times and the android platform of 9 is works fine, it's only happened after install the apk, if I restart the app after the installation, it works fine.

**Comment by ping996** [comment]
I also has another one of two isolate port communication, if I click the back button of the android system, the flutter app will back to invisible and the app process is working, when I click it and on focus again, the main isolate will not receive message successfully, my question is the main isolate and the second on of I created will destroy? 
When I click it on focus again, the main of the dart will executed again, so it will create the second isolate again, are there there isoalte now? the main and the two of I created?
